### PR TITLE
Make pip/setup.py install dependencies

### PR DESCRIPTION
**Pull request by hobarrera**
Use `setuptools` with `install_requires` so that running pip or `setup.py` directly will install all dependencies.

### CODE REVIEW

**Comment by hobarrera** [comment]
Hint: By doing this, you can also run `pip install -e .` with the virtualenv active, so that it picks up the local version as if it were installed (so you could end up simplifying `bin/xandikos`.

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ cache:
 install:
   - pip install pip --upgrade
   - sudo apt-get install -qq libneon27-dev
-  - pip install -r requirements.txt
+    python setup.py develop
```
Comment (jelmer) [inline .travis.yml]: This lacks a leading dash.

Comment (hobarrera) [inline .travis.yml]: My bad, fixed.

File: setup.py [hunk 2]
```diff
@@ -31,7 +31,9 @@
       author_email="jelmer@jelmer.uk",
       license="GNU GPLv3 or later",
       url="https://www.jelmer.uk/projects/xandikos",
-      install_requires=['icalendar', 'dulwich', 'defusedxml'],
+      install_requires=[
+          open('requirements.txt').readlines()
```
Comment (jelmer) [inline setup.py]: Can we drop requirements.txt instead? Reading https://packaging.python.org/requirements/#requirements-files it serves a different role, and I don't think it's actually necessary in this case.

Comment (hobarrera) [inline setup.py]: TBH, for applications (eg: no libraries), I tend to prefer [this workflow](https://www.kennethreitz.org/essays/a-better-pip-workflow).

But it's really your call. `requirements.txt` is more of a de facto convention than anything else.

Comment (jelmer) [inline setup.py]: I'd prefer to keep things simple; I don't use pip myself, but rely on Debian packages so am unlikely to keep a requirements.txt up to date. There shouldn't be a need for xandikos to have more than a handful of dependencies (if that changes, we can revisit this). 

Can you drop requirements.txt for the moment, and update the reference to it in README.md to 
use 'setup.py develop' ?

Comment (hobarrera) [inline setup.py]: Great, it's all the same to me, TBH. :)
PR updated.

**Comment by jelmer** [comment]
Merged, thanks!

**Pull request closed without merging**